pkg/apis/kops/validation/aws.go

```diff
@@ -56,6 +56,32 @@ func awsValidateInstanceGroup(ig *kops.InstanceGroup, cloud awsup.AWSCloud) fiel
 		allErrs = append(allErrs, awsValidateMixedInstancesPolicy(field.NewPath("spec", "mixedInstancesPolicy"), ig.Spec.MixedInstancesPolicy, ig, cloud)...)
 	}
 
+	if ig.Spec.InstanceMetadata != nil {
+		allErrs = append(allErrs, awsValidateInstanceMetadata(field.NewPath("spec", "instanceMetadata"), ig.Spec.InstanceMetadata)...)
+	}
+
+	return allErrs
+}
+
+func awsValidateInstanceMetadata(fieldPath *field.Path, instanceMetadata *kops.InstanceMetadataOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	httpTokens := fi.StringValue(instanceMetadata.HTTPTokens)
+	httpPutResponseLimit := fi.Int64Value(instanceMetadata.HTTPPutResponseHopLimit)
+
+	// we do not allow disabling InstanceMetadata since some k8s and kops components require InstanceMetadata
+	if httpTokens != "required" && httpTokens != "optional" {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("httpTokens"), instanceMetadata.HTTPTokens,
+			"Invalid HTTPTokens value. Possible values are 'required' or 'optional'. See https://docs.aws.amazon.com/cli/latest/reference/ec2/modify-instance-metadata-options.html"+
+				" for more details"))
+	}
+
+	if httpPutResponseLimit <= 0 || httpPutResponseLimit > 64 {
```

Comment:
Would 1 actually work with kops? Don't we have containers that use the IMDS?

Reply:
are you referencing to IMDv1 or httpPutResponseLimit?

Reply:
I believe your referring to the httpPutHopReponseLimit if i m right. I was not aware of the fact that we had some containers using IMDS in kops. I will cross check this with aws documentation.

Reply:
I believe the IMDS is a special link local address as stated in the aws docs. This may not be an issue. But I am not sure whether overlay networks like Calico will subtract the ttl before sending it out of the cluster. 

I believe AWS kept the default ttl 1 to mitigate issues against misconfigured NAT and router instances. 
[IMDv2](https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/)

Reply:
I refer to `HttpPutResponseHopLimit` yes. TTL of 1 would mean no routing. So pods with hostNetwork will work, but pods using overlay IPs will not. CNIs using ENIs (lyft, cilium-eni, vpc) will probably also work, while those using overlay networks will not.

Reply:
thanks for this insight.

@olemarkus correct me if I wrong here. when i grepped for 169.254.169.254 ip, i see it being used only here [https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/resources/addons/metadata-proxy.addons.k8s.io/v0.1.12.yaml](https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/resources/addons/metadata-proxy.addons.k8s.io/v0.1.12.yaml) 
in the context of a container. hostNetworking is enabled for this container. I think a ttl of 1 should work in this case, if this is the only container in kops which is accessing the instance metadata service.

Reply:
Fairly sure there are more containers using it, but I cannot think of anything using overlay network.
Easy enough to change later if this fails.

Reply:
got it. makes sense.